### PR TITLE
po_lines_location.sql

### PR DESCRIPTION
**Pull request by arthur-aguilera**
This ports the derived table po_lines_locations to Metadb

### CODE REVIEW

File: po_lines_location [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        jsonb_extract_path_text(locations.data, 'quantity') AS pol_loc_qty,
+        jsonb_extract_path_text(locations.data, 'quantityElectronic') AS pol_loc_qty_elec,
+        jsonb_extract_path_text(locations.data, 'quantityPhysical') AS pol_loc_quant_phys
+    FROM
+        folio_orders.po_line AS pol
+        CROSS JOIN jsonb_array_elements(jsonb_extract_path(jsonb, 'locations')) AS locations (data)
+)
+SELECT
+    pol_location.pol_id,
+    pol_location.pol_loc_id,
```
Comment (adoerrer) [inline po_lines_location:19]: You might want to cast pol_loc_id as well to uuid data type.

File: po_lines_location [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    pol_location.pol_loc_qty,
+    pol_location.pol_loc_qty_elec,
+    pol_location.pol_loc_quant_phys,
```
Comment (adoerrer) [inline po_lines_location:22]: I think all of them could be casted  to `int`.

File: po_lines_location [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+-- Create a local table for Purchase Order Line Locations including location quantity and type.
+
+CREATE TABLE po_lines_locations AS
+WITH pol_location AS (
+    SELECT
+        pol.id::uuid AS pol_id,
+        jsonb_extract_path_text(locations.data, 'locationId') AS pol_loc_id,
+        jsonb_extract_path_text(locations.data, 'quantity') AS pol_loc_qty,
+        jsonb_extract_path_text(locations.data, 'quantityElectronic') AS pol_loc_qty_elec,
+        jsonb_extract_path_text(locations.data, 'quantityPhysical') AS pol_loc_quant_phys
```
Comment (adoerrer) [inline po_lines_location:12]: Perhaps `pol_loc_quant_phys` could named as `pol_loc_qty_phys` just for naming consistency.

**Review by adoerrer — CHANGES_REQUESTED** [review]
Thank you, Arthur! The query looks and works great overall, but I've some minor suggestions on the data type casts and a formal naming consistency.

Beside that you need to add the derived table query to the [runlist.txt](https://github.com/arthur-aguilera/folio-analytics/blob/main/sql_metadb/derived_tables/runlist.txt)  So it can  be run automatically.


- [x] Query runs
- [x] Data output makes sense
- [x] Data types look correct
- [x] Query logic seems correct
- [x] Column names make sense and are spelled correctly
- [x] Query syntax and style are appropriate (e.g., correct use of double quotes, lower case for data types)
For derived tables:
- [x] query has indexes on columns
- [x] vacuum analyze has been added to the bottom
- [x] table appears in runlist.txt at an appropriate point and with correct file name

**Review by adoerrer — APPROVED** [review]
Now that looks and works great. Thank you, Arthur!

**Comment by nassibnassar** [comment]
Could we have a complete description for the pull request:  https://github.com/folio-org/folio-analytics/blob/main/CONTRIBUTING.md#2-commits-and-pull-requests

**Comment by adoerrer** [comment]
> Could we have a complete description for the pull request: https://github.com/folio-org/folio-analytics/blob/main/CONTRIBUTING.md#2-commits-and-pull-requests

I think this is my bad. I just advised arthur to enter the related issue. 
@arthur-aguilera 
Please, add something like: This ports the derived table po_lines_locations to Metadb
Thanks!

**Pull request merged**